GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -212,6 +212,20 @@ public override FileSystemResult WritePlaceholderDirectory(string relativePath)
             }
         }
 
+        public override FileSystemResult DehydrateFolder(string relativePath)
+        {
+            FileSystemResult result = new FileSystemResult(FSResult.Ok, 0);
+            GitIndexProjection.PathSparseState sparseState = this.FileSystemCallbacks.GitIndexProjection.GetFolderPathSparseState(relativePath);
+
+            if (sparseState == GitIndexProjection.PathSparseState.Included)
+            {
+                result = this.WritePlaceholderDirectory(relativePath);
+                this.FileSystemCallbacks.OnPlaceholderFolderCreated(relativePath, string.Empty);

```

Comment:
Could you add a comment here explaining why we need to do this when the folder is Included?